tempto-core/src/main/java/io/prestosql/tempto/internal/query/NonPoolingJdbcDataSource.java

```diff
@@ -46,7 +48,8 @@ public Connection getConnection()
             throws SQLException
     {
         Properties properties = prepareConnectionProperties();
-        return driver.connect(jdbcParamsState.url, properties);
+        Connection connection = driver.connect(jdbcParamsState.url, properties);
+        return requireNonNull(connection, "connection is null");
```

Comment:
Does it make sense to quote `jdbcParamsState.url`?
```suggestion
        return requireNonNull(connection, () -> format("connection is null for '%s'", jdbcParamsState.url));
```